### PR TITLE
DOC: special: add a full docstring for `pdtr`

### PR DESCRIPTION
**Pull request by person142**
## Reference issue

None

#### What does this implement/fix?

A complete docstring is added for `pdtr`.

#### Additional information

In particular, add

- a more complete definition
- a reference
- parameters/returns/see also sections
- examples

### CODE REVIEW

**Comment by person142** [comment]
Hm failure due to the warning when using real `k` being upgraded to an error by the tests. Would be fixed by https://github.com/scipy/scipy/issues/10855, or I guess I could suppress the warning manually in the example; feels a little weird in an example though.

File: scipy/special/add_newdocs.py [hunk 1]
```diff
@@ -6997,14 +6997,59 @@ def add_newdoc(name, doc):
...
+    :math:`k`. More concretely, this works out to be [1]_
+
+    .. math::
+
+       \exp(-m) \sum_{j = 0}^{\lfloor{k}\rfloor} \frac{m^j}{m!}.
+
+    Parameters
+    ----------
+    k : array_like
+        Nonngegative real argument
```
Comment (tylerjereddy) [inline scipy/special/add_newdocs.py]: "Nonngegative" typo, I think

File: scipy/special/add_newdocs.py [hunk 2]
```diff
@@ -6997,14 +6997,59 @@ def add_newdoc(name, doc):
...
+    References
+    ----------
+    .. [1] https://en.wikipedia.org/wiki/Poisson_distribution
+
+    Examples
+    --------
+    >>> import scipy.special as sc
+
+    It is a cumulative distribution function, so it converges to 1
+    monotonially as `k` goes to infinity.
```
Comment (tylerjereddy) [inline scipy/special/add_newdocs.py]: "monotonically"?

**Comment by person142** [comment]
I'm going to wait for @rlucas7 to fix https://github.com/scipy/scipy/issues/10855 before finishing this PR.

**Comment by person142** [comment]
Ok, https://github.com/scipy/scipy/pull/10956 has been merged, so this PR can proceed. Just rebased and addressed @tylerjereddy's review comments.

File: scipy/special/add_newdocs.py [hunk 3]
```diff
@@ -7037,14 +7037,58 @@ def add_newdoc(name, doc):
...
+    :math:`k`. More concretely, this works out to be [1]_
+
+    .. math::
+
+       \exp(-m) \sum_{j = 0}^{\lfloor{k}\rfloor} \frac{m^j}{m!}.
+
+    Parameters
+    ----------
+    k : array_like
+        Nonnegegative real argument
```
Comment (tylerjereddy) [inline scipy/special/add_newdocs.py]:
```suggestion
        Nonnegative real argument
```

**Review by tylerjereddy — COMMENTED** [review]
1 typo still left -- ready to merge after that is fixed

File: scipy/special/add_newdocs.py (hunk 3 above)
Comment (person142) [reply scipy/special/add_newdocs.py]: Har I really can’t spell apparently.

**Comment by rgommers** [comment]
Looks good now, merged. Thanks @person142

**Pull request merged**